src/Workspaces/Core/Portable/Shared/Utilities/ProducerConsumer.cs

```diff
@@ -116,6 +116,30 @@ internal static class ProducerConsumer<TItem>
             cancellationToken);
     }
 
+    /// <summary>
+    /// Version of RunAsync that will process <paramref name="source"/> in parallel.
+    /// </summary>
+    public static Task RunParallelAsync<TSource, TArgs>(
+        IAsyncEnumerable<TSource> source,
```

Comment:
this overload operates on an IAsyncEnumerable source.  I can't find ways to make these overloads without a lot of duplication of the bodies unfortunately.